Assets/Scripts/Player/PlayerInventory.cs

```diff
@@ -202,16 +207,41 @@ void PickUpBuddy( BuddyStats buddyStats )
 			_backBuddy.hiddenBuddy = buddyStats;
 			_backBuddy.hiddenBuddy.gameObject.SetActive( false );
 
+			if ( _backBuddyHappinessRoutine != null )
+			{
+				StopCoroutine( _backBuddyHappinessRoutine );
+			}
+
+			_backBuddyHappinessRoutine = StartCoroutine( BackBuddyHappinessRoutine( _backBuddy.hiddenBuddy ) );
 			_isCarryingBuddy = true;
 		}
 	}
 
-	void GiveResource( BuddyStats buddyStats )
+	IEnumerator BackBuddyHappinessRoutine( BuddyStats buddyStats )
 	{
-		buddyStats.GiveResource( (ResourceData) _heldResources[_resourceIndex] );
-		_inventory[_heldResources[_resourceIndex]]--;
+		while ( true )
+		{
+			// Wait time should be first so that the player cannot spam pick up/put down
+			// to get happiness for free
+			yield return new WaitForSeconds( _backBuddyHappinessWaitTime );
 
-		UpdateResourceList();
+			buddyStats.AdjustHappiness( _backBuddyHappinessIncrement );
+			buddyStats.RecalculateStat();
```

Comment:
I don;t think it matters much. Even in discrete chunks it's so small it's nearly unnoticeable.